README.md

```diff
@@ -5,19 +5,73 @@
 
+* Model - The model is the chef, it should manage the critical aspects of the application. One of my favorite tasks in a Rails application is working with the model files. This is where you can be very expressive with the custom algorithms that you want to utilize and you also have direct access to the specific database record. The logic of your application should mainly reside in the model files.
+
+* Controller - The controller is the waiter. In the same way that the waiter takes the order from the customer to the chef and then brings the food to the table, the controller transmits data requests from the user to the model, and then delivers data that is rendered in the view to the user.
+
+* View - The view is the table. A table shouldn't do anything besides sit there and hold the food when it is delivered. In like manner the views should not contain any programming logic, they should simply render what the controller sends it. One of the top issues I discover when I review a Rails application that has a large number of bugs is that the developer integrated too much logic directly into the view.
+
+## Routing, File Naming Conventions, and Data Flow
+
+Rails was created with the concept of convention over configuration and this holds true for how the MVC structure was setup. View files correspond directly to controller files, which speak directly with models. As an example, imagine that you have a blog that has a database table called ```posts```. You will have the following set of files:
+
+* A ```post.rb``` model file that will contain: validations, database relationships, callbacks, and any custom logic for posts.
+
+* A ```posts_controller.rb``` file that will have methods that will manage data flow for the Post behavior, this will include the full set of CRUD features, the standard methods are: index, new, create, show, edit, update, and destroy.
+
+* A ```views/``` directory that will contain a corresponding view for each of the pages that the end user will access. For a CRUD based model, a few of the standard views would include: an index view to show all records, a show page that shows a specific record, and then new and edit pages that both render a form.
+
+## Request Flow
+![MVC Request Flow](http://reif.io/lib/flatiron/mvc-flow.jpg)
+
+## Roles and Responsibilities
+
+### Models
+
+At the end of the day the model file is a Ruby class. If it has a corresponding database table it will inherit from the ActiveRecord::Base class, which means that it has access to a number of methods that assist in working with the database. However, you can treat it like a regular Ruby class, allowing you to create methods, data attributes, and everything else that you would want to do in a class file. In a typical model file you will find: custom database query scopes, default settings for database columns, validations, model-to-model relationships, callbacks, and custom algorithms related to that model.
```

Comment:
The last sentence "in a typical model file...." there is a ton of jargon that they will have no idea means. Because of that, students are just going to gloss over this. I would make it a bit more general. Something like "A typical model contains your domain logic. This means, everything from database queries, relationships and all the other magic that would happen in a kitchen that's hidden from the waiters and the patrons of the restaurant"